api/handler/seattle.go

```diff
@@ -0,0 +1,116 @@
+package handler
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sort"
+	"spd-lookup/api/data"
+	"strings"
+)
+
+// SeattleOfficerMetadata is the handler function for retrieving SPD metadata
+func (h *Handler) SeattleOfficerMetadata(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(http.StatusOK)
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(h.db.SeattleOfficerMetadata())
+}
+
+// SeattleStrictMatch is the handler function for retrieving SPD officers with a strict match
+func (h *Handler) SeattleStrictMatch(w http.ResponseWriter, r *http.Request) {
+	badge, firstName, lastName := r.URL.Query().Get("badge"), r.URL.Query().Get("first_name"), r.URL.Query().Get("last_name")
+
+	if badge != "" {
+		h.seattleGetOfficerByBadge(badge, w)
+		return
+	} else if firstName != "" || lastName != "" {
+		h.seattleGetOfficersByName(firstName, lastName, w)
+		return
+	} else {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(fmt.Sprintf("at least one of the following parameters must be provided: badge, first_name, last_name")))
+	}
+}
+
+func (h *Handler) seattleGetOfficerByBadge(badge string, w http.ResponseWriter) {
+	ofc, err := h.db.SeattleGetOfficerByBadge(badge)
+
+	if err != nil {
+		if err.Error() == "no rows in result set" {
+			w.WriteHeader(http.StatusOK)
+			w.Header().Set("Content-Type", "application/json")
+			json.NewEncoder(w).Encode([]*data.SeattleOfficer{})
+			return
+		}
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte(fmt.Sprintf("error getting officer: %s", err)))
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode([]*data.SeattleOfficer{ofc})
+}
+
+func (h *Handler) seattleGetOfficersByName(firstName, lastName string, w http.ResponseWriter) {
+	if firstName == "" {
```

Comment:
Is there a specific case here (or upstream) for handling both `firstName` and `lastName` being empty? Because it seems that, were that the case, all records would be returned which doesn't seem ideal :sweat_smile:

Reply:
Yup, this function is called from line 27 in the same module (the SeattleStrictMatch function)

`else if firstName != "" || lastName != "" {`